pkg/alertmanager/multitenant.go

```diff
@@ -270,33 +270,24 @@ func (am *MultitenantAlertmanager) updateConfigs(now time.Time) error {
 }
 
 // poll the configuration server. Not re-entrant.
-func (am *MultitenantAlertmanager) poll() (map[string]configs.View, error) {
-	configID := am.latestConfig
-	cfgs, err := am.configsAPI.GetAlerts(context.Background(), configID)
+func (am *MultitenantAlertmanager) poll() (map[string]alerts.AlertConfigDesc, error) {
+	cfgs, err := am.store.ListAlertConfigs(context.Background())
 	if err != nil {
 		level.Warn(util.Logger).Log("msg", "MultitenantAlertmanager: configs server poll failed", "err", err)
 		return nil, err
 	}
-	am.latestMutex.Lock()
-	am.latestConfig = cfgs.GetLatestConfigID()
-	am.latestMutex.Unlock()
-	return cfgs.Configs, nil
+	return cfgs, nil
 }
 
-func (am *MultitenantAlertmanager) addNewConfigs(cfgs map[string]configs.View) {
+func (am *MultitenantAlertmanager) addNewConfigs(cfgs map[string]alerts.AlertConfigDesc) {
 	// TODO: instrument how many configs we have, both valid & invalid.
 	level.Debug(util.Logger).Log("msg", "adding configurations", "num_configs", len(cfgs))
-	for userID, config := range cfgs {
-		if config.IsDeleted() {
```

Comment:
Before this PR, we would have removed a config no more existing. In this PR, we filter out removed users in the store, which means the `cfgs` input parameter here does not contain removed users, but then we never reconcile `am.alertmanagers` and `am.cfgs` to remove users no more in the `cfgs`.

Moreover the `deleteUser()` is not called by anyone now.

Reply:
I updated this function to include logic to do this. I also have removed deleteUser in order to avoid obfuscating when the mutex is being used.